projects/core/src/button/button.element.spec.ts

```diff
@@ -279,17 +281,62 @@ describe('button element', () => {
       expect(component.disabled).not.toBeTruthy();
     });
 
-    it('should stay disabled when loadingState changes to default', async () => {
+    it('should go to disabled set before loadingState changed to non-default, when it changes back', async () => {
       await componentIsStable(component);
-      component.disabled = true;
+      component.disabled = true; // should be restored, once loadingState changes back to default
+
+      component.loadingState = ClrLoadingState.default;
+      await componentIsStable(component);
+
+      component.loadingState = ClrLoadingState.loading;
+      await componentIsStable(component);
+      expect(component.disabled).toBeTruthy();
+
+      component.loadingState = ClrLoadingState.success;
+      await componentIsStable(component);
+      expect(component.disabled).toBeTruthy();
+
+      component.loadingState = ClrLoadingState.default;
+      await componentIsStable(component);
+      expect(component.disabled).toBeTruthy();
+    });
+
+    it('should go to enabled set while loadingState was non-default, when it changes back', async () => {
+      await componentIsStable(component);
+      component.disabled = true; // should NOT be restored, as it is changed while in non-default loadingState
```

Comment:
I'm confused by this test, by default the loadingState is the default loading state

Reply:
Can you keep the existing names and implementations for these unit tests. I think it's helpful to have tests where the implicit behavior happens - ie, no one ever set the disabled status before mucking with the loading states